components/zcash_note_encryption/src/lib.rs

```diff
@@ -490,6 +512,14 @@ pub fn try_output_recovery_with_ock<D: Domain, Output: ShieldedOutput<D>>(
         domain.parse_note_plaintext_without_memo_ovk(&pk_d, &esk, output.epk(), &plaintext)?;
     let memo = domain.extract_memo(&plaintext);
 
+    // ZIP 212: Check that the esk provided to this function is consistent with the esk we
+    // can derive from the note.
```

Comment:
Does this have to be able to handle pre-ZIP212 plaintexts?

Reply:
Pre-ZIP 212 note plaintexts will return `None` from `D::derive_esk()`.